sycl/test/extensions/free_function_kernels/free_function_host_compiler.cpp

```diff
@@ -0,0 +1,31 @@
+// RUN: %clangxx -fsycl -fsycl-host-compiler=g++ %s
+// RUN: %clangxx -fsycl -fno-sycl-unnamed-lambda %s
```

Comment:
I think this should exercise three cases: 1) default compilation, 2) disabled unnamed lambda support, and 3) third party host compiler (which has unnamed lambda support disabled)
```suggestion
// RUN: %clangxx -fsycl %s
// RUN: %clangxx -fsycl -fno-sycl-unnamed-lambda %s
// RUN: %clangxx -fsycl -fsycl-host-compiler=g++ %s
```